### DISCUSSION TITLE
model_construct() doesn't apply nested model with discriminator

### DISCUSSION THREAD

**Discussion by fspillner** [pydantic/pydantic#10035] [Question]
Hi, 

Given the following Pydantic model:

```python
from pydantic import *
from typing import Literal

class OptionA(BaseModel):
    option_type: Literal["option-a"] = "option-a"
    option_a: str

class OptionB(BaseModel):
    option_type: Literal["option-b"] = "option-b"
    option_b: str

Option = OptionA | OptionB

class Model(BaseModel):
    option: Option = Field(discriminator="option_type")
```

I am able to construct the model from a dictionary using:

```python
Model(**dict(
    option=dict(
        option_type="option-a",
        option_a="A",
    )
))
```

This results in:

```Model(option=OptionA(option_type='option-a', option_a='A'))```

However, using `Model.model_construct()` yields an unexpected result:

```python
Model.model_construct(**dict(
    option=dict(
        option_type="option-a",
        option_a="A",
    )
))
```

This results in:

```Model(option={'option_type': 'option-a', 'option_a': 'A'})```

Am I doing something incorrectly? Is there a workaround besides manually constructing the model?

Thank you for your help / advice 🙂🙏

**Comment by Viicos** [comment]
This is the expected behavior of [`model_construct`](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_construct): it assumes the provided data is already validated, and that means potentials type coercion:

```python
from pydantic import BaseModel


class Sub(BaseModel):
    foo: int


class Model(BaseModel):
    sub: Sub


m = Model.model_construct(sub={"foo": 1})
#> Model(sub={'foo': 1}
```
